docs/standard/serialization/binaryformatter-migration-guide/functionality-reference.md

```diff
@@ -0,0 +1,89 @@
+}
+```
+
+The previous class is translated by the C# compiler to:
+
+```csharp
+[Serializable]
+internal class PropertySample
+{
+    private string <Name>k__BackingField;
+
+    public string Name
+    {
+        get
+        {
+            return <Name>k__BackingField;
+        }
+        set
+        {
+            <Name>k__BackingField = value;
+        }
+    }
+}
+```
+
+In this case, `<Name>k__BackingField` is **the name of the member that `BinaryFormatter` uses in the serialized payload**. It's not possible to use `nameof` or any other C# operator to get this name.
+
+The [ISerializable](/dotnet/api/system.runtime.serialization.iserializable) interface comes with [GetObjectData(SerializationInfo info, StreamingContext context)](/dotnet/api/system.runtime.serialization.iserializable.getobjectdata) method that allows the users to control the names, by using one of the [SerializationInfo.AddValue](/dotnet/api/system.runtime.serialization.serializationinfo.addvalue) methods.
+
+```csharp
+// Note lack of any special attribute.
+public void GetObjectData(SerializationInfo info, StreamingContext context)
+{
+    info.AddValue("Name", this.Name);
+}
+```
+
+If such customization has been applied, the information needs to be provided during deserialization as well. That's possible by using the **serialization constructor**, where all values are read from `SerializationInfo` with one of the `Get` methods it provides.
+
+```csharp
+private PropertySample(SerializationInfo info, StreamingContext context)
+{
+    this.Name = info.GetString("Name");
+}
+```
+
+> [!NOTE]
+> The `nameof` operator was purposely not used here, as the payload can be persisted and the property can get renamed at a later time. So even if it gets renamed (say to `FirstName` because you decide to also introduce a `LastName` property), to remain backward compatible, the serialization should still use the old name that could have been persisted somewhere.
+
+## Serialization binder
+
+It's recommended to use [SerializationBinder](/dotnet/api/system.runtime.serialization.serializationbinder) to control class loading and mandate what class to load. That minimizes security vulnerabilities (so only allowed types get loaded, even if the attacker modifies the payload to deserialize and load something else).
+
+Using this type requires inheriting from it and overriding the [Type BindToType(string assemblyName, string typeName)](/dotnet/api/system.runtime.serialization.serializationbinder.bindtotype#system-runtime-serialization-serializationbinder-bindtotype(system-string-system-string)) method.
+
+If your code uses a custom type that derives from `SerializationBinder`, it's likely that the list of serializable types is a **closed set**. If not, finding the list of all types that can get serialized and deserialized requires studying all the usages of `BinaryFormatter` in source code. **Knowing that list is crucial to determining how to move forward with the migration.**
```

Comment:
> **Knowing that list is crucial to determining how to move forward with the migration.**

This feels a bit like a non-sequitur, given that we don't talk about migration here.

Should we reword this?

```suggestion
Ideally the list of serializable types is closed set because it means you know which types can be instantiated which will help reduce security vulnerabilities. 
```